components/script/dom/servoparser/mod.rs

```diff
@@ -648,13 +648,14 @@ impl FetchResponseListener for ParserContext {
                     parser.parse_sync();
                 }
             },
-            Some(ContentType(Mime(TopLevel::Text, SubLevel::Xml, _))) => {}, // Handle text/xml
+            // Handle xml (text/xml, application/xml, application/xhtml+xml)
+            Some(ContentType(Mime(ref toplevel @ _, ref sublevel @ _, _)))
+            if (
+                ((toplevel == &TopLevel::Application) && (sublevel.as_str() == "xhtml+xml")) ||
+                ((toplevel == &TopLevel::Application) && (sublevel == &SubLevel::Xml)) ||
+                ((toplevel == &TopLevel::Text) && (sublevel == &SubLevel::Xml))
+            ) => {},
```

Comment:
I'm actually not quite fond of using a match guard for this purpose; I'd rather have multiple match arms that matches exactly the TopLevel and SubLevel types.